Community insights

Arielle-Bennett opened the pull request:
### Summary

Will close #4507 



### List of changes proposed in this PR (pull-request)

* Add a chapter landing page on understanding your community
* Add a first version of a summary table for approaches to community assessment and analysis 
* Add a first version of the "running a community insights report" subchapter 
* Will cross-ref existing community report examples as case studies 

### What should a reviewer concentrate their feedback on?

- [ ] *Lorem ipsum dolor sit amet, consectetur adipiscing.*
- [ ] Everything looks ok?


### Acknowledging contributors


- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/the-turing-way/the-turing-way/blob/main/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/the-turing-way/the-turing-way/blob/main/README.md#contributors) in the README file: <!-- replace this text with the GitHub IDs of any new contributors -->

Arielle-Bennett committed:
Add 'Understanding Your Community' chapter

This chapter discusses approaches for assessing community strengths and weaknesses, and offers strategies for focused development.

Arielle-Bennett committed:
Add community insights summary document

This document summarizes various community assessment processes, outlining methodologies, positives, negatives, and resources for each type of assessment.

Arielle-Bennett committed:
Revise community insight report chapter structure and content

Arielle-Bennett committed:
Revise Community Insight Report chapter for clarity

Expanded the chapter on Community Insight Reports to include detailed processes, considerations, benefits, and communication strategies. Enhanced clarity and structure for better understanding.

EstherPlomp commented on book/website/collaboration/understanding-community/community-insight-report.md:
```diff
@@ -0,0 +1,134 @@
+Regardless, the rest of this section contains questions and prompts to help you document the answers to various aspects of producing this type of report which you might need to cover as part of such a submission and which should be communicated to the community. 
+
+### How will you tell people?
+Communicating with the wider community about the process of producing a Community Insight Report is critical. The exercise in and of itself can be a great community engagement strategy and you will need people to know it is happening if you want them to participate in the interviews.  
```
```suggestion
Communicating with the wider community about the process of producing a Community Insight Report is critical.
The exercise in and of itself can be a great community engagement strategy and you will need people to know it is happening if you want them to participate in the interviews.  
```

EstherPlomp reviewed (changes requested):
Hi @Arielle-Bennett - thanks for adding this! I think it is a great suggestion to use the SSI projects as case studies later on. 

I don't think this is in the TOC yet? I just wanted to check that before approving. Other than that I think this looks great, thank you!

Arielle-Bennett committed:
Create community insight report example for JupyterHub

Added an example community insight report for JupyterHub.

Arielle-Bennett committed:
Add community understanding section to myst.yml

Arielle-Bennett committed:
Apply suggestion from @EstherPlomp

Co-authored-by: Esther Plomp <46314469+EstherPlomp@users.noreply.github.com>

Arielle-Bennett committed:
Add vojh-report reference to myst.yml

Arielle-Bennett committed:
Update myst.yml

KirstieJane commented on book/website/myst.yml:
```diff
@@ -814,6 +815,15 @@ project:
+           - file: collaboration/understanding-community/community-insight-report.md
+             title: Running a Community Insight Report
+           - file: collaboration/understanding-community/community-insight-report-example.md
+             title: Example Community Insight Report: Voices of JupyterHub
```
```suggestion
        - file: collaboration/understanding-community.md
          title: Understanding Your Community
          children:
          - file: collaboration/understanding-community/community-insights-summary.md
            title: Summary of Different Community Assessment Processes
          - file: collaboration/understanding-community/community-insight-report.md
            title: Running a Community Insight Report
          - file: collaboration/understanding-community/community-insight-report-example.md
            title: Example Community Insight Report: Voices of JupyterHub
```

KirstieJane replied:
I think this whitespace change should let the CI run.... I also think I have write access to this PR so I'm going to push the change now!

KirstieJane committed:
Fix whitespace error in myst.yml file

KirstieJane committed:
fix whitespace in myst.yml

KirstieJane committed:
Fixing whitespace again!

KirstieJane commented on book/website/myst.yml:
```diff
@@ -23,6 +23,7 @@ project:
   references:
     myst-spec: https://mystmd.org/spec
     myst-guide: https://mystmd.org/guide/
+    vojh-report: https://voicesofjupyterhub.orgmycology.com/
```
```suggestion
    vojh-report: "https://voicesofjupyterhub.orgmycology.com"
```

KirstieJane committed:
Put the voices of jupyterhub link in quotes

JupyterBook seems to be parsing the url at ".org" where really it should keep going! The organization name is orgmycology!

KirstieJane commented on book/website/myst.yml:
```diff
@@ -23,6 +23,7 @@ project:
   references:
     myst-spec: https://mystmd.org/spec
     myst-guide: https://mystmd.org/guide/
+    vojh-report: "https://voicesofjupyterhub.orgmycology.com"
```
```suggestion
    vojh-report: "https://jupyterbook.org/"
```

KirstieJane committed:
Remove link to voices of jupyterhub report

Just to see if this is the problem!

KirstieJane commented on book/website/myst.yml:
```diff
@@ -814,6 +815,15 @@ project:
+            - file: collaboration/understanding-community/community-insight-report.md
+              title: Running a Community Insight Report
+            - file: collaboration/understanding-community/community-insight-report-example.md
+              title: Example Community Insight Report: Voices of JupyterHub
```
```suggestion
              title: "Example Community Insight Report: Voices of JupyterHub"
```

KirstieJane commented on book/website/myst.yml:
```diff
@@ -23,6 +23,7 @@ project:
   references:
     myst-spec: https://mystmd.org/spec
     myst-guide: https://mystmd.org/guide/
+    vojh-report: "https://jupyterbook.org/"
```
```suggestion
    vojh-report: "https://voicesofjupyterhub.orgmycology.com/"
```

KirstieJane committed:
quote title with colon in it!

Co-authored-by: Kirstie Whitaker <kirstie.whitaker@berkeley.edu>

KirstieJane commented on book/website/collaboration/understanding-community/community-insights-summary.md:
```diff
@@ -0,0 +1,13 @@
+# Summary of Different Community Assessment Processes
+
+The table below offers a brief overview of some of the different techniques and methods you can use to assess your community. 
+It is not exhaustive, and more contributions are always welcome. 
```
I wonder if we can expand this table to have a few more links to existing chapters in the book? Or even sections that are about to be featured in this new chapter?

KirstieJane commented on book/website/collaboration/understanding-community/community-insight-report.md:
```diff
@@ -0,0 +1,135 @@
+
+In general, it is best to plan to recruit using a mixture of methods, with a general call to the community highlighting the areas of focus, and specific invites to a cross-section of the community aligned with your goals for the report to ensure a diversity of participants, backgrounds, and roles. 
+
+:::warning
```
```suggestion
:::{warning}
```

KirstieJane committed:
Fix admonition formatting

KirstieJane commented:
Fixed the build - woo! 

Turns out the problem was putting a `:` in the title of a sub chapter! Shout out to @JimMadge for helping me find that in today's collab cafe!

I think that means @EstherPlomp - you can dismiss your request for changes!

KirstieJane commented:
There are a few little bitty edits that I'd like to make and one substantial question that I'd love input on.

The chapter is written such that the Voices of JupyterHub report appears right at the end... so I _personally_ find the chapter a bit vague. I think I'd rather have links to VoJ as an _exemplar_ project all the way through.... but I think I understand why @Arielle-Bennett wrote the chapter as kinda "theory-first" and then practice! If we have more insight reports to feature then we aren't over indexing on this first one??

What do others think? Would it be an easier read if we shouted out the VoJ report nearer the start of this chapter??

KirstieJane committed:
fix conflict

KirstieJane committed:
update a few links

KirstieJane committed:
put voices of jupyterhub report in quotes

bsipocz commented on book/website/myst.yml:
```diff
@@ -30,7 +30,8 @@ project:
     myst-spec: https://mystmd.org/spec
-    myst-guide: https://mystmd.org/guide/
+    myst-guide: https://mystmd.org/guide
+    vojh-report: https://voicesofjupyterhub.orgmycology.com/index
```
try without index so the ast can be picked up properly

```suggestion
    vojh-report: https://voicesofjupyterhub.orgmycology.com/
```

bsipocz reviewed (commented):
@KirstieJane - I think my suggestion should do the fix you were asking for in `#infrasrtructure`

KirstieJane committed:
Update myst.yml

Co-authored-by: Brigitta Sipőcz <b.sipocz@gmail.com>

KirstieJane commented:
Just committed to check - but I think I tried adding index last week and it didn't solve the problem 😩

sgibson91 commented on book/website/collaboration/understanding-community/community-insight-report-example.md:
```diff
@@ -0,0 +1,9 @@
+(cl-community-insight-report-example)=
+# An Example Community Insight Report: Voices of JupyterHub
+
+Embedded below is the introduction of an example of how (#cl-community-insight-report) can be conducted. 
```
```suggestion
Embedded below is the introduction of an example of how [](#cl-understanding-community-insight) can be conducted. 
```

This is a malformed reference and also the original target you put doesn't exist. I've taken a guess at what you intended, but double check it please?

sgibson91 replied:
```suggestion
.... the introduction of an example of how (#cl-community-insight-report) can be conducted. 
```

You won't be able to embed, so this sentence (maybe the whole section?) will need restructuring/rewriting.

sgibson91 commented on book/website/myst.yml:
```diff
@@ -30,7 +30,8 @@ project:
     myst-spec: https://mystmd.org/spec
-    myst-guide: https://mystmd.org/guide/
+    myst-guide: https://mystmd.org/guide
+    vojh-report: https://voicesofjupyterhub.orgmycology.com/
```
```suggestion
    vojh-report: https://voicesofjupyterhub.orgmycology.com
```

sgibson91 committed:
fix: try stripping the trailing / from the vojh url as that is not present when you visit the page itself

Co-authored-by: Sarah Gibson <44771837+sgibson91@users.noreply.github.com>

JimMadge commented:
```⛔️ myst.yml Unable to load reference "vojh-report" as MyST cross-references or intersphinx```

Strange difference in build behaviour locally and in GitHub CI & Netlify.
Possibly the behaviour of MyST is different in CI/headless environments and the requests are getting refused.

JimMadge commented on book/website/collaboration/understanding-community/community-insight-report-example.md:
```diff
@@ -0,0 +1,9 @@
+This example was produced by Organizational Mycology on behalf of the JupyterHub project. 
+Read the full report at: https://voicesofjupyterhub.orgmycology.com/
+
+![Read the Voices of JupyterHub Report Introduction](xref:vojh-report#vojhreport-introduction)
```
It might be good here to add some decoration to show that this is the embedded section from the report.

JimMadge commented on book/website/collaboration/understanding-community/community-insight-report.md:
```diff
@@ -0,0 +1,137 @@
+### Who to include and how to approach them? 
+Depending on your goals and areas of interest for the Community Insight Report, you may wish to solicit interviewees in different ways: 
+
+* **Identifying a list of potential interviewees at the beginning**
```
The list items here might work better as headings,

sgibson91 commented:
I have confirmed the discrepency between local and CI builds. The VOJH report is hosted on Cloudflare Pages (https://github.com/orgmycology/vojhreport/blob/main/.github/workflows/main.yml#L41) and this service has a load of WAFs and bot fighters that block access from datacenter IPs which services like GitHub Actions and Netlify will use.

The quickest and simplest way to resolve this is to not embed the content at all. Just provide a link for readers to click through to.

sgibson91 commented on book/website/collaboration/understanding-community/community-insight-report-example.md:
```diff
@@ -0,0 +1,9 @@
+This example was produced by Organizational Mycology on behalf of the JupyterHub project. 
+Read the full report at: https://voicesofjupyterhub.orgmycology.com/
+
+![Read the Voices of JupyterHub Report Introduction](xref:vojh-report#vojhreport-introduction)
```
```suggestion
```

We can't embed this because the site is hosted on Cloudflare Pages which blocks access to datacenter IPs from services like GitHub Actions and Netlify, hence this will _always_ break our CI.

JimMadge replied:
It does seem that way. @Arielle-Bennett do you think there would be interest in changing the hosting? As this will prevent everyone from embedding sections of the report.
